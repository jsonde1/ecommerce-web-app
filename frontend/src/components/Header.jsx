import { useRef } from "react";
import { Collapse } from "bootstrap";
import { NavLink } from "react-router-dom";
import logo from "../assets/DFBay.png";
import Ticker from "./Ticker";
import ReactTicker from "./ReactTicker";

const Header = ({ user, logOut }) => {
  const collapseMenu = useRef();
  user = user || {};
  const shown = Object.keys(user).length > 0 ? `nav-link` : `nav-link disabled`;
  const toggleCollapse = () => {
    const collapseEle = collapseMenu.current;
    const bsCollapse =
      Collapse.getInstance(collapseEle) ?? new Collapse(collapseEle);
    bsCollapse.toggle();
  };

  return (
    <>
      <header>
        <nav className="navbar navbar-expand-lg navbar-dark shadow">
          <div className="container-fluid ">
            <a
              href="/"
              className="navbar-brand"
              target="_blank"
              rel="noreferrer"
            >
              <img src={logo} alt="DFBay Logo" width="50" />
            </a>
            <button
              className="navbar-toggler"
              type="button"
              data-bs-toggle="collapse"
              data-bs-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
            >
              <span className="navbar-toggler-icon"></span>
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                <li className="nav-item">
                  <NavLink
                    to="/"
                    className={({ isActive }) =>
                      isActive ? `nav-link active` : `nav-link`
                    }
                  >
                    Search
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink
                    to="/listings"
                    className={({ isActive }) =>
                      isActive ? `nav-link active` : `nav-link`
                    }
                  >
                    Listings
                  </NavLink>
                </li>
                <li className="nav-item">
                  {Object.keys(user).length === 0 && (
                    <NavLink to="/loginregister" className="nav-link">
                      Login/Register
                    </NavLink>
                  )}
                </li>
                <li className="nav-item">
                  <NavLink
                    to={
                      user.UserType == "standard"
                        ? `/user/${user.id}`
                        : `/admin`
                    }
                    className={({ isActive }) =>
                      isActive ? `${shown} active` : shown
                    }
                  >
                    Dashboard
                  </NavLink>
                </li>
                <li className="nav-item">
                  {Object.keys(user).length > 0 && (
                    <a href="/" className="nav-link" onClick={logOut}>
                      Log Out
                    </a>
                  )}
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <ReactTicker />
      </header>
    </>
  );
};

export default Header;
