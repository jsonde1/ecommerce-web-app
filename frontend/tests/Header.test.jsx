import { render } from "@testing-library/react";
import Header from "../src/components/Header.jsx";
import { MemoryRouter } from "react-router-dom";
import { describe, test, expect } from "vitest";

test(`Header matches snapshot when no user is logged in`, () => {
  const user = {};
  expect(
    render(
      <MemoryRouter>
        <Header user={user} />
      </MemoryRouter>
    )
  ).toMatchSnapshot();
});
