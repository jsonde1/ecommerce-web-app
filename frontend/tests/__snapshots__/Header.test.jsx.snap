// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header matches snapshot when no user is logged in 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="bg-secondary"
      >
        <nav
          class="navbar navbar-expand-lg navbar-dark "
        >
          <div
            class="container-fluid "
          >
            <a
              class="navbar-brand"
              href="/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="DFBay Logo"
                src="/src/assets/react.svg"
                width="50"
              />
            </a>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-lg-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="nav-link active"
                    href="/"
                  >
                    Search
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/listings"
                  >
                    Listings
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/loginregister"
                  >
                    Login/Register
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link disabled"
                    href="/admin"
                  >
                    Dashboard
                  </a>
                </li>
                <li
                  class="nav-item"
                />
              </ul>
            </div>
          </div>
        </nav>
        <div
          class="ticker-container"
        >
          <div
            class="ticker-wrapper"
          >
            <div
              class="ticker-transition"
            >
              <div
                class="ticker-item"
              >
                GBP/USD 
              </div>
              <div
                class="ticker-item"
              >
                GBP/EUR 
              </div>
              <div
                class="ticker-item"
              >
                GBP/JPY 
              </div>
              <div
                class="ticker-item"
              >
                GBP/AUD 
              </div>
              <div
                class="ticker-item"
              >
                GBP/CAD 
              </div>
              <div
                class="ticker-item"
              >
                GBP/NGN 
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="bg-secondary"
    >
      <nav
        class="navbar navbar-expand-lg navbar-dark "
      >
        <div
          class="container-fluid "
        >
          <a
            class="navbar-brand"
            href="/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="DFBay Logo"
              src="/src/assets/react.svg"
              width="50"
            />
          </a>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarSupportedContent"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarSupportedContent"
          >
            <ul
              class="navbar-nav me-auto mb-2 mb-lg-0"
            >
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="nav-link active"
                  href="/"
                >
                  Search
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/listings"
                >
                  Listings
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/loginregister"
                >
                  Login/Register
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link disabled"
                  href="/admin"
                >
                  Dashboard
                </a>
              </li>
              <li
                class="nav-item"
              />
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="ticker-container"
      >
        <div
          class="ticker-wrapper"
        >
          <div
            class="ticker-transition"
          >
            <div
              class="ticker-item"
            >
              GBP/USD 
            </div>
            <div
              class="ticker-item"
            >
              GBP/EUR 
            </div>
            <div
              class="ticker-item"
            >
              GBP/JPY 
            </div>
            <div
              class="ticker-item"
            >
              GBP/AUD 
            </div>
            <div
              class="ticker-item"
            >
              GBP/CAD 
            </div>
            <div
              class="ticker-item"
            >
              GBP/NGN 
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
